project/src/main/webapp/script.js

```diff
@@ -400,3 +407,29 @@ function registerUserByToken() {
   }
   fetch('/register?idToken=' + googleUser.getAuthResponse().id_token, {method: 'POST'});
 }
+
+/** Sends feedback that includes the place the user chose. */
+function sendUserChoiceAsFeedback() {
+  let indexOfPlaceUserChose = getCheckedValueByElementId('chosen-place-form',
+      'Please select the place that you chose.');
+  if (indexOfPlaceUserChose <= recommendedPlaces.length) {
+    updateUserFeedback(recommendedPlaces, recommendedPlaces[indexOfPlaceUserChose].placeId,
+        false /** user tried again */)
+  }
+}
+
+/** Updates user feedback in the database, by using the feedback servlet. */
+function updateUserFeedback(recommendedPlaces, chosenPlaceId, tryAgain) {
```

Comment:
We already have a "getUserIdToken" function in our js, you should use it instead of lines 423-425. And do :
      `idToken=${getUserIdToken()}`
Just like the fetchFromQuery request.

Notice that this way the check in the Servlet should be:
If `userIdToken.isEmpty()` instead of `userIdToken == null`, just like the treatment in the Query Servlet.
I think it is also better for consistency to have it the same.

Reply:
That's right (we have the function), but we don't want to get rid of the check (423-425), because we don't want to call the servlet if we don't have a user (even if he checks if it's empty or not, there is no reason to add a redundant call)

Reply:
That check was a part of the function... the exact same lines

Reply:
(You want the current function to return, it's not enough that the function you call returns, if you just call the function and not do the check you'll still have a redundant call to the servlet)